app/webpacker/controllers/order_info_controller.js

```diff
@@ -43,6 +48,32 @@ export default class extends Controller {
     return { note: data }
   }
 
+  confirmRemove(params) {
+    this.displayAlert((confirmation) => {
+      if (confirmation) {
+        this.makeApiCall(params)
+      }
+    })
+  }
+
+  displayAlert(callback) {
```

Comment:
Yes I agree. I thought of doing that here but I think it can be more effectively extracted into a separate component if I work on converting the orders cancellation functionality into Stimulus first. They share the same delete dialog box that would help me see what parts of it they share and what parts they don't and write the component accordingly. 

What do you think? I can open up an issue for it and begin working on converting that part into stimulus and extract the alert part into a seperate component.

Reply:
Yes, I follow you, open an issue then, to not forget it! 🙏